File: Arrays/LeetCode78.java
import java.util.ArrayList;
import java.util.List;

class LeetCode78 {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(new ArrayList<>());

        for(int ele: nums){
            int len = ans.size();
            for(int i = 0; i<len; i++){
                ArrayList<Integer> temp = new ArrayList<>(ans.get(i));
                temp.add(ele);
                ans.add(temp);
                
            }
            for(List<Integer> a : ans){
                System.out.println(a);
            }
        }
        return ans;
    }
    public static void main(String[] args) {
        LeetCode78 obj = new LeetCode78();
        int[] num = {1,2,3};
        obj.subsets(num);
    }
}

File: Backtracking/LeetCode78.py
class Solution:
    def subsets(self, nums):
        Ans = []
        n = len(nums)

        # Helper function
        def Helper(op, start_index):
            # base case
            if start_index == len(nums):
                return
            
            Ans.append(op)

            Helper(op,start_index+1)
            op.append(nums[start_index])
            Helper(op,start_index+1)

            print(op)
        
        Helper([], 0)
        return Ans
obj = Solution()
print(obj.subsets([1,2,3]))